### PR TITLE
Adds mail to cargo

### PR DESCRIPTION
**Pull request by SandPoot**
## About The Pull Request
***Port of [tgstation#58146](https://github.com/tgstation/tgstation/pull/58146)***

This ports Kiwi's Mail system to cargo as a form of passive job responsibilities to cargo and the cargo shuttle. The original PR was listed here: https://github.com/tgstation/tgstation/pull/55279.
This is not a one to one recreation of the original PR, however, as I did some tweaking and changes to the system since it was first done mostly in terms of economy balance and cargo balance, and I'm by no means perfect on this, but I think this will keep the game's balance from absolutely exploding.

So, from the top.
The Economy Subsystem now fires an additional check, which will will add to a new counter called mail_waiting. This a passive stack of how much mail is waiting between cargo shuttle trips. The healthier the station's economy is, as well as based on the number of living crew, the more mail will come in. Market inflation or market crash events can lower the amount of mail waiting for you as a result. Previously this was a clean 1 mail per every 5 minutes, but this lead to very little mail coming in and not much engagement with the rest of the crew, so this was amped up as a result. The most mail you can get in 5 minutes is set by the define MAX_MAIL_PER_FIVE_MINUTES, and may be subject to changes before this gets undrafted.

Mail, if mail is waiting, will arrive on a mail tub on the cargo shuttle. Cargo techs can open it up as a crate subtype, and distribute mail to crew. Crew who have addressed mail are the only people who can open their letters, preventing despicable mail fraud. Remember kids, mail fraud is illegal, and illegal things are a crime. Anyway, mail can be delivered the old fashion way (by hand), sent through disposals via the destination system (Destination Taggers), or by loudly screaming into the radio that Jacob Ladders needs to pick up his DAMN MAIL.

Jobs can have specific goodies that can appear in the mail, based on their occupation. For example, the captain can sometimes get a bottle of champagne or cigars on occasion, all determined by a pickweight list system. In addition, there is a generic goodies list that consists of small cash bonuses for the owner. The old version of this list in the original PR was really, really REALLLLY high. Like, 10,000 credit spacecash bills were added to the game mostly as a way of debugging inflation, not to be handed out that easily. Still, the cash option exists as an occasional nice bonus to certain jobs, perhaps helping to soften a paycheck shaped hole in your heart.

In addition, the original PR introduced sprites for the mailcrate, mailbag, the actual mail letters themselves, and a few others. I went ahead and did a handful of sprite tweaks, completely redoing the mail tub icon_states and doing some tweaks to the individual letters. In addition, the letters now use the item_scaling element, similar to TCG cards, as there's a good number of them that spawn and otherwise they create a good deal of clutter.

## Why It's Good For The Game
Having played with this a few months ago, this added a really unique dynamic to cargo where cargo actually had more of a role running resources between the shuttle and the crew, with a nice bit of dynamic gameplay involving how to actually get players their mail, especially when the crew gets occasional rewards for staying alive to receive it.

While the first PR was certainly good, it did need a bit of balance tweaking from then and now, especially after doing the work with in-game prices and rescaling involving the cargo shuttle. This brings much of the pricing and scaling in line with current costs, without the risk of too much gamer loot getting thrown into the system, albeit some of the pickweights and list items may need a second opinion.

Giving some of these things some new sprites and tweaks will help improve the visual style of much of this addition as well off from the original testmerges.

## Changelog
🆑
add: The mail system from Kiwistation has been ported.
add: Cargo will gradually recieve mail crates over time, which contains letters and packages to and for the crew. Cargo should make an attempt to deliver people's mail.
add: The International Spess Mail Legion has finally convinced the universe of the evils of mail fraud.
/:cl:

### CODE REVIEW

**Comment by SandPoot** [comment]
It is incentivized to suggest for loots

File: code/controllers/subsystem/economy.dm [hunk 1]
```diff
@@ -70,7 +78,8 @@ SUBSYSTEM_DEF(economy)
 	for(var/account in bank_accounts)
 		var/datum/bank_account/bank_account = account
 		bank_account.payday(1)
-
+	var/effective_mailcount = round(living_player_count()/1) // (inflation_value - 0.5)) //More mail at low inflation, and vis versa.
```
Comment (SandPoot) [inline code/controllers/subsystem/economy.dm]: I need someone to suggest something to replace (inflation_value - 0.5).

**Comment by deathride58** [comment]
Mail deliveries are rly funny, +support

**Comment by TripleShades** [comment]
> It is incentivized to suggest for loots

Items that are holiday-relevant? When applicable.

**Comment by SandPoot** [comment]
> > It is incentivized to suggest for loots
> 
> Items that are holiday-relevant? When applicable.

This does sound possible

**Review by deathride58 — APPROVED** [review]
`You've got mail!`

**Pull request merged**